app/lib/request.rb

```diff
@@ -94,7 +94,7 @@ def key_id
   end
 
   def timeout
-    { connect: 10, read: 10, write: 10 }
+    { connect: nil, read: 10, write: 10 }
```

Comment:
`Timeout.timeout(nil)` yields the block immediately, without spawning any threads